### ISSUE TITLE
Anchor element misparsed

### ISSUE THREAD

**Issue by vouillon** [aantron/markup.ml#3]
The following string is misparsed:
```
let s = "<!doctype html><a href=\"/\">ab</a>\n<p>cd</p>"
```
The paragraph element gets included inside a second anchor element.
Indeed, the following piece of code:
```
s |> string |> parse_html |> signals |> pretty_print |> write_html |> to_channel stdout;;
```
ouputs:
```
<!DOCTYPE html>
<html>
  <head>
  </head>
  <body>
    <a href="/">
      ab
    </a>
    <a href="/">
      <p>
        cd
      </p>
    </a>
  </body>
</html>
```
This is with version 0.6 in opam.

**Comment by aantron** [comment]
Thank you, looking into these.

**Comment by aantron** [comment]
Should be fixed, along with #4 and #5. Output is now:

```html
<!DOCTYPE html>
<html>
  <head>
  </head>
  <body>
    <a href="/">
      ab
    </a>
    <p>
      cd
    </p>
  </body>
</html>
```

HTML is specified using an ugly state machine that is pretty resistant to (straightforward) analysis, so I'm planning to write a very large number of tests. I started by improving Bisect_ppx together with Leonid Rozenberg (see e.g. https://github.com/rleonid/bisect_ppx/pull/56). Hopefully this will allow to catch the large proportion of parser bugs.

In the meantime, please keep the bug reports coming :)
